### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by serprex**
Introduce 2 failing tests
Refactor tests to use unwrap
Refactor tokens to use enums
Flush queue with while-let

I intend to follow up with fixing the failing tests I've introduced & improving parse to handle multidigit numbers

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -8,47 +8,50 @@ pub fn parse(code: &str) -> Result<Vec<RPNToken>, String> {
...
                 output.push(queue.pop().unwrap());
                 queue.push(rpnt)
             } else {
-                if tok == ')' {
+                if tokop == Operator::RPAREN {
                     for t in queue.clone() {
-                        if t.value != '(' {
+                        if t != RPNToken::Operator(Operator::LPAREN) {
                             let v = queue.pop().unwrap();
-                            if v.value != '(' || v.value != ')' {
```
Comment (serprex) [inline src/parse.rs]: This is a tautology, so I changed || to &&

**Comment by squiidz** [comment]
great job ! thanks

**Comment by squiidz** [comment]
Are you ready to merge ?

**Comment by serprex** [comment]
All tests now passing, ready to merge

**Comment by squiidz** [comment]
Best pull request ever 👍

**Pull request merged**